scripts/build-jorje.js

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env node
+/* 
+ * Automates the process of updating the Language Server Jar from a local Jorje repo.
+ * 
+ * Assumptions:
+ * 0. You have shelljs installed globally using `npm install -g shelljs`.
+ * 1. You have a local Jorje repo, have saved the keystore to sign the Jar, and have 
+ *    saved the paths for those + the keystore to your bash profile. Reset VS Code once set.
+ * 
+ * Ex:  export JORJE_DEV_DIR=~/Git-Repositories/apex-jorje
+        export SFDC_KEYSTORE=~/KEYS/sfdc.jks
+        export SFDC_KEYPASS=PASS #where PASS is the password for the keystore
+ */
+
+const process = require('process');
+const shell = require('shelljs');
+const fs = require('fs');
+const util = require('util');
+const { checkJorjeDirectory, checkSigningAbility } = require('./validation-utils');
+const logger = require('./logger-util');
+
+//Input Variables
+const needSigning=process.argv[2];
+
+//Messages
+const LOG_HEADER = '='.repeat(80);
+const SECTION_HEADER = '-'.repeat(80);
+const NO_JORJE_MSG = `Apex Jorje source repository not found, verify the path of the environment variable JORJE_DEV_DIR: %s. \n`;
+const NO_FILE_MSG = `File not found, verify the path of the environment variable %s: %s\n`;
+const BUILD_MSG = `Building and signing apex-jorje project\n  from: %s...\n`;
+const FIND_MSG = `Searching for artifacts\n     In: %s...\n`;
+const COPY_MSG = `Copying: %s\n     To: %s\n`;
+const COPY_ERR = `Failed to copy LSP jar: %s. Error: %s\n`;
+const SIGN_ERR = `Failed to sign jar: %s\n`;
+const SIGN_INFO = `Ignore the certificate signer expired errors. We are ok with these.`;
+
+//Directories
+const CURRENT_DIR=process.cwd();
+let JORJE_DEV_DIR='';
+let SFDC_KEYSTORE='';
+let SFDC_KEYPASS='';
+const JORJE_DEST_DIR=`${CURRENT_DIR}/packages/salesforcedx-vscode-apex/out`;
+const JORJE_DEST_PATH=`${JORJE_DEST_DIR}/apex-jorje-lsp.jar`;
+
+function verifyPaths() {
+  console.log(LOG_HEADER);
+  try {
+    JORJE_DEV_DIR = checkJorjeDirectory();
+    fs.statSync(JORJE_DEV_DIR).isDirectory();
```

Comment:
@randi274 The return value is not referenced, so this check has no effect, i.e. a non-directory value can be used and it will fail further on.

Reply:
It actually does fail if it's not a valid directory (which is an interesting way write a function for a library), but it does lead to another question of: what if it's a directory, but the wrong one? I think I can catch for that later on, and add some better error handling.
![Screen Shot 2021-12-06 at 4 14 27 PM](https://user-images.githubusercontent.com/9795193/144923626-b91a098f-a239-4e2d-9093-63b875983f3d.png)

Reply:
Yes, an invalid file path will cause the check to fail, but if it is a valid file, but not a valid directory then it will continue and fail at **process.chdir(JORJE_DEV_DIR)** in **buildLSP()**.

Reply:
I moved things around a bit and think that issue should be resolved now; let me know if you'd like me to be more explicit about handling it though!